### PR TITLE
Cmdct 4125 - refactor report page footer

### PR DESCRIPTION
**Pull request by britt-mo**
### Description
Refactored report page footer to be more like that of MFP. Logic for `previous` and the two kinds of `continue` buttons are in their own variables and called from inside the `return` statement. Also simplified styling.

### Related ticket(s)

CMDCT-4125

---
### How to test
compare any environment (local, dev, val, etc) to the cloudfront link and see there shouldn't be any difference in the UI of the report page footer.

### Notes

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/components/reports/ReportPageFooter.tsx [hunk 1]
```diff
@@ -20,56 +20,57 @@ export const ReportPageFooter = ({
     report?.formTemplate.flatRoutes,
     report?.formTemplate.basePath
   );
-  const hidePrevious = previousRoute === "/mcpar" || previousRoute === "/mlr";
 
-  const { userIsAdmin, userIsReadOnly } = useStore().user ?? {};
+  const { userIsAdmin, userIsReadOnly, userIsEndUser } = useStore().user ?? {};
   const isAdminUserType = userIsAdmin || userIsReadOnly;
-  const formIsDisabled = isAdminUserType && !form?.editableByAdmins;
+  const hidePrevious = previousRoute === "/mcpar" || previousRoute === "/mlr";
```
Comment (gmrabian) [inline services/ui-src/src/components/reports/ReportPageFooter.tsx:26]: do you mind adding `/naaar` to this while you're here? 😄

File: services/ui-src/src/components/reports/ReportPageFooter.tsx [hunk 2]
```diff
@@ -20,56 +20,57 @@ export const ReportPageFooter = ({
...
+  const formIsDisabled =
+    (isAdminUserType && !form?.editableByAdmins) ||
+    (userIsEndUser && reportWithSubmittedStatus);
+  const isReadOnly = !form?.id || formIsDisabled;
+
+  const prevButton = (
+    <Button
+      onClick={() => navigate(previousRoute)}
+      variant="outline"
+      leftIcon={<Image src={previousIcon} alt="" sx={sx.arrowIcon} />}
```
Comment (gmrabian) [inline services/ui-src/src/components/reports/ReportPageFooter.tsx:37]: we lost the alt text on the icons for this button and the rest

File: services/ui-src/src/components/reports/ReportPageFooter.tsx [hunk 3]
```diff
@@ -86,24 +87,21 @@ interface Props {
...
     width: "8.25rem",
   },
+  nextButton: {
+    minWidth: "8.25rem",
+    marginLeft: "auto",
+  },
   praStatement: {
     fontSize: "0.875rem",
-    paddingTop: "2rem",
+    marginTop: "1rem",
```
Comment (gmrabian) [inline services/ui-src/src/components/reports/ReportPageFooter.tsx:105]: I noticed the gap between PRA and the continue button in Figma is huge! I know 1rem is what we have in MFP, so I'm not sure what to follow, just throwing it out there. I also think 1rem will look better when the previous button is gone.


Screenshot 2024-11-19 at 9 25 57 AM

**Review by gmrabian — COMMENTED** [review]
Two small requests!

This reminded me as well that we want to have the bar separating the report questions from the footer as part of the footer as well. I will make a ticket for that!
Screenshot 2024-11-19 at 9 28 10 AM

File: services/ui-src/src/components/reports/ReportPageFooter.tsx (hunk 2 above)
Comment (britt-mo) [reply services/ui-src/src/components/reports/ReportPageFooter.tsx:37]: Good catch

File: services/ui-src/src/constants.ts [hunk 4]
```diff
@@ -79,3 +79,6 @@ export enum States {
 // TIMEOUT PARAMS
 export const IDLE_WINDOW = 30 * 60 * 1000; // ms
 export const PROMPT_AT = 29 * 60 * 1000; //ms
+
+// routes
+export const ROUTES = ["/mcpar", "/mlr", "/naaar"];
```
Comment (gmrabian) [inline services/ui-src/src/constants.ts:84]: nit: i'd give this a more descriptive name like `DASHBOARD_ROUTES` or something

**Review by gmrabian — APPROVED** [review]

File: services/ui-src/src/constants.ts (hunk 4 above)
Comment (britt-mo) [reply services/ui-src/src/constants.ts:84]: that's fair

**Review by bangbay-bluetiger — APPROVED** [review]

**Pull request merged**